src/components/filtermenu/filtermenu.js

```diff
@@ -103,35 +103,27 @@ function onInputCommand(e) {
     }
 }
 function saveValues(context, settings, settingsKey, setfilters) {
-    let elems = context.querySelectorAll('.simpleFilter');
-
     // Video type
     const videoTypes = [];
-    elems = context.querySelectorAll('.chkVideoTypeFilter');
-
-    for (let i = 0, length = elems.length; i < length; i++) {
-        if (elems[i].checked) {
-            videoTypes.push(elems[i].getAttribute('data-filter'));
+    for (const elem of context.querySelectorAll('.chkVideoTypeFilter')) {
```

Comment:
`forEach` is more portable (for legacy) than `for..of`. :thinking:

Reply:
It sounds like we should setup some eslint rules to prevent us from using some of these language features that make generated code more complex since it has came up a few times now. I can also disable related rules on the sonar side as needed.

Reply:
Tried `eslint-plugin-no-for-of-loops`: 147 errors :sweat_smile:

Reply:
Do you want to change the usages here or leave it as is?